app/services/anno_auth_validator.rb

```diff
@@ -0,0 +1,43 @@
+require 'json'
+require 'net/http'
+require 'jwt'
+
+class AnnoAuthValidator
+
+    def initialize(authHeader, canvas)
+        @authHeader = authHeader
+        @canvas = canvas
+      end
+
+    # Verfiying from remote destination that the token sent has the same group ID for the canvas and that the user has access
+    def authorize
+        begin
```

Comment:
I don't think you need to explicitly use `begin` here - if you just `rescue` at the end of a method it works.

Reply:
You can clean up your control flow a little bit here:

```ruby
def authorize
  token = ...
  data = ...
  data["group_id"] == token.first["group_id"] && data["has_access"] == "true" # will return your boolean

  rescue StandardError => _
    false
end
```

Reply:
FYI, the reason we rescue `StandardError` and not `Exception` can be found [here](http://docs.honeybadger.io/ruby/exceptions/what-are-ruby-exceptions.html#rescuing-all-exceptions-the-wrong-way).